avm/res/web/hosting-environment/import.bicep

```diff
@@ -0,0 +1,324 @@
+
+@description('Optional. Scale factor for frontends.')
+param frontEndScaleFactor int = 15
+
+@description('Optional. Specifies which endpoints to serve internally in the Virtual Network for the App Service Environment. - None, Web, Publishing, Web,Publishing. "None" Exposes the ASE-hosted apps on an internet-accessible IP address.')
+@allowed([
+  'None'
+  'Web'
+  'Publishing'
+  'Web, Publishing'
+])
+param internalLoadBalancingMode string = 'None'
+
+@description('Optional. Number of IP SSL addresses reserved for the App Service Environment. Cannot be used when kind is set to ASEv3.')
+param ipsslAddressCount int = 0
+
+@description('Optional. Frontend VM size. Cannot be used when kind is set to ASEv3.')
+@allowed([
+  ''
+  'Medium'
+  'Large'
+  'ExtraLarge'
+  'Standard_D2'
+  'Standard_D3'
+  'Standard_D4'
+  'Standard_D1_V2'
+  'Standard_D2_V2'
+  'Standard_D3_V2'
+  'Standard_D4_V2'
+])
+param multiSize string = ''
+
+@description('Optional. Property to enable and disable new private endpoint connection creation on ASE. Ignored when kind is set to ASEv2.')
+param allowNewPrivateEndpointConnections bool = false
+
+@description('Optional. Property to enable and disable FTP on ASEV3. Ignored when kind is set to ASEv2.')
+param ftpEnabled bool = false
+
+@description('Optional. Customer provided Inbound IP Address. Only able to be set on Ase create. Ignored when kind is set to ASEv2.')
+param inboundIpAddressOverride string = ''
+
+@description('Optional. Property to enable and disable Remote Debug on ASEv3. Ignored when kind is set to ASEv2.')
+param remoteDebugEnabled bool = false
+
+@description('Optional. Specify preference for when and how the planned maintenance is applied.')
+@allowed([
+  'Early'
+  'Late'
+  'Manual'
+  'None'
+])
+param upgradePreference string = 'None'
+
+@description('Required. ResourceId for the subnet.')
+param subnetResourceId string
+
+@description('Optional. User added IP ranges to whitelist on ASE DB. Cannot be used with \'kind\' `ASEv3`.')
+param userWhitelistedIpRanges array = []
+
+@description('Optional. Switch to make the App Service Environment zone redundant. If enabled, the minimum App Service plan instance count will be three, otherwise 1. If enabled, the `dedicatedHostCount` must be set to `-1`.')
+param zoneRedundant bool = false
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities) ? {
```

Comment:
Does this happen to show warnings locally? If so, I'd suggest to apply the fix @krbar implemented in this PR for other modules: #1238

Reply:
Yeh it does actually.  Warning BCP187: The property "identity" does not exist in the resource or type definition.

THe solution in that PR doesnt address this one though. Im assuming this property can be removed from the resource block based on this: https://learn.microsoft.com/en-us/azure/templates/microsoft.web/hostingenvironments?pivots=deployment-language-bicep

Reply:
Hey @tsc-buddy it seems to be supported though: https://learn.microsoft.com/en-us/azure/app-service/environment/how-to-custom-domain-suffix?pivots=experience-azp#managed-identity ... interestingly enough

Reply:
Would you consider that a docs (first link) and bicep error bug potentially? Keen to get your thoughts and I can then try to pin down a PM.

Reply:
It may be a bug, yes. The only thing that can give us certainty is deploying the module with the identity in there (ignoring the warning) and checking if it's set on the Azure Service as per the docs.

Reply:
Ok, test running now. Will keep you posted. Thanks Alex.

Reply:
@AlexanderSehr tested this overnight and the microsoft.web doc for bicep I believe is incorrect and flags the warning for that identity property as discussed. Yet the below link is correct and managed identity is supported.

So with that can we resolve this one and I can work out offline how to engage PG to address this for Bicep?

https://learn.microsoft.com/en-us/azure/app-service/environment/how-to-custom-domain-suffix?pivots=experience-azp#managed-identity

Reply:
Sounds like a plan, yes. Thanks @tsc-buddy